dropshot/src/handler.rs

```diff
@@ -205,10 +195,16 @@ macro_rules! impl_extractor_for_tuple {
     #[async_trait]
     impl< $($T: Extractor + 'static,)* > Extractor for ($($T,)*)
     {
-        async fn from_request<Context: ServerContext>(_rqctx: Arc<RequestContext<Context>>)
+
+        // This `allow` prevents a false positive from clippy. See -
+        // https://github.com/rust-lang/rust-clippy/issues/4637
+        #[allow(clippy::eval_order_dependence)]
+        async fn from_request<Context: ServerContext>(_rqctx: &mut RequestContext<Context>)
             -> Result<( $($T,)* ), HttpError>
         {
-            futures::try_join!($($T::from_request(Arc::clone(&_rqctx)),)*)
+            Ok(($(
+                $T::from_request(_rqctx).await?
+            ,)*))
         }
```

Comment:
I just saw that paste is already a dependency, so we could use it to do something like this. 

```suggestion
        #[allow(non_snake_case)]
        async fn from_request<Context: ServerContext>(_rqctx: &mut RequestContext<Context>)
            -> Result<( $($T,)* ), HttpError>
        {
            paste! {
                $(
                let [<v_ $T>] = $T::from_request(_rqctx).await?;
                )*

                Ok(($(
                    [<v_ $T>]
                ,)*))
            }
        }
```

It still requires a lint ... plus it's kind of messy :p

Reply:
I'm not sure that's better; tbh I think I prefer the original. A comment saying "tuple initialization order is well-defined, so the lint is a false positive" would IMO be the right direction.

Reply:
100% agreed. I just updated the comment. Thanks, @smklein